gourmet/tests/test_reccard.py

```diff
@@ -229,29 +294,25 @@ def test_undo_save_sensitivity (rc):
         except:
             print('rc.widgets_changed_since_save',rc.widgets_changed_since_save)
             raise
-        if VERBOSE: print('DONE TESTING %s'%widget)
-
-rg = GourmetRecipeManager.RecGui()
-rg.newRecCard()
-while Gtk.events_pending(): Gtk.main_iteration()
-rec_id,rec_card = list(rg.rc.items())[0]
-
-try:
-    test_ing_editing(rec_card)
-    print('Ing Editing works!')
-    test_ing_undo(rec_card)
-    print('Ing Undo works!')
-    test_undo_save_sensitivity(rec_card)
-    print('Undo properly sensitizes save widget.')
-    test_ing_group_editing(rec_card)
-    print('Ing Group Editing works.')
-except:
-    import traceback; traceback.print_exc()
-    Gtk.main()
-else:
-    rec_card.hide()
-    import sys
-    sys.exit()
+        print_("DONE TESTING", widget)
 
 
+with TemporaryDirectory(prefix='gourmet_', suffix='_test_reccard') as tmpdir:
```

Comment:
Should this be in an `if __name__ == '__main__':` block or something? Putting it like this means the module will run its own tests on import, and then the test runner (nose) will run them again.

Reply:
Oh, I get it - the test functions need a reccard instance. I think this block should still be in a test function or method, itself, so that it's not trying to run the tests when it's imported.

Reply:
Certainly! But first I'd like the whole shebang to work as intended. However, the structure of the application has changed quite a bit since these tests were implemented, as you noted earlier.

And so I'm currently attempting to make it work, and then I will restructure the tests :)

I think they needn't be interdependent, and each could have their own reccard instance.

Thanks for checking though, I'd have missed that point!